.github/workflows/docker.yml

```diff
@@ -0,0 +1,64 @@
+name: Build & Push Docker Images
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+    - develop
+
+jobs:
+  pre_job:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Login to DockerHub
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v2
+        with:
+            username: ${{ secrets.DOCKERHUB_USERNAME }}
+            password: ${{ secrets.DOCKERHUB_TOKEN }}
```

Comment:
Added these secrets in the PyBaMM repo. Let me know if everything works.

Reply:
This CI is working on my fork [here](https://github.com/arjxn-py/PyBaMM/actions/runs/6266390975/job/17017264690?pr=6).
Can you please once confirm if secrets are set how we want them to?

Reply:
Yes! We'll only be able to test it once this PR is merged. The secrets are not passed down to PRs from forks for security reasons.